lib/windows/app/actions/appReloadDialogActions.js

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2015 - 2017, Nordic Semiconductor ASA
+ *
+ * All rights reserved.
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Indicates that the AppReloadDialog should be shown. Apps will typically
+ * dispatch this action to show the AppReloadDialog after errors occur to
+ * help user reloading the app in order to fix the issues.
+ */
+export const APP_RELOAD_DIALOG_SHOW = 'APP_RELOAD_DIALOG_SHOW';
+
+/**
+ * Indicates that the AppReloadDialog should be hidden. Apps will typically
+ * dispatch this action to hide the AppReloadDialog after reloading the app
+ * or being cancelled.
+ */
+export const APP_RELOAD_DIALOG_HIDE = 'APP_RELOAD_DIALOG_HIDE';
+
+export function showDialog(message, reload) {
+    return {
+        type: APP_RELOAD_DIALOG_SHOW,
+        message,
+        reload,
+    };
+}
+
+export function hideDialog() {
+    return {
+        type: APP_RELOAD_DIALOG_HIDE,
+    };
+}
+
+// Reload app
+export function reloadApp() {
+    return (dispatch, getState) => {
+        getState().core.appReloadDialog.reload();
```

Comment:
We need to look at some other solution. State is not a place to store functions, lets move this to the container.